### DISCUSSION TITLE
VX3 - Datapoints for Mennekes Amtron Wallbox

### DISCUSSION THREAD

**Discussion by HDMatGIT** [open3e/open3e#140] [Q&A]
Hallo,
erstmal vielen Dank an die Macher dieses tollen Tools! Dank Eurer Aktivitäten habe ich fast alle Daten über den CAN-Bus lokal zur Verfügung und verarbeite diese via MQTT in openHAB 4.
Ich habe eine PV Anlage komplett von Viessmann: PV 9,6 kWp, 10 kW Batterie, VX3, E380 und eine Wallbox von Mennekes.
Leider habe ich noch nicht herausgefunden, wie ich die Daten (hauptsächlich die Energie Daten) der Wallbox erhalten kann. Die Wallbox ist in die Anlage (über EEBUS) eingebunden und im ViCare gibt es dazu auch einen Button. Dort kann ich dann auch einstellen ob ich mit voller Power (11kW) oder "grün" Laden will. Dort wird auch angezeigt mit wieviel Watt gerade geladen wird. Ich würde gern den Energieverbrauch der Wallbox separat vom Verbrauch im gesamten Haus erfassen und an openHAB senden. Gibt es dazu einen entsprechenden Decoder oder ist das über (welche?) Datenpunkte aus dem VX3 aus lesbar?    
Vielen Dank schon mal!

**Comment by MyHomeMyData** [comment]
Ich habe eine sehr ähnliche Anlage mit E380, VX3, Vitocal und Mennekes Wallbox. Allerdings ist die Wallbox nicht in die VX3 eingebunden. Das gab es nicht als meine Anlage in Betrieb genommen wurde.

Ich steuere die Wallbox per direkter Modbus -Verbindung. Nutze dazu ioBroker. Funktioniert bestens, habe alle Daten und kann das Stromangebot flexibel steuern. 

Hast Du depictSystem gemacht? Vielleicht gibt es da vielversprechende Datenpunkte (*wallbox*)?

Gruß, Jürgen

**Comment by HDMatGIT** [comment]
Hallo Jürgen, danke für die rasche Antwort und Deine Hinweise. Ja depictSystem habe ich gemacht, aber für die Wallbox habe ich leider nix gefunden. Dann muss ich mich demnächst wohl mal mit dem Modbus beschäftigen...
Herzliche Grüße
Hans-Dieter

**Comment by achgut** [comment]
Hallo
Ich habe eine VX3, E380 und eine Siemens Wallbox. Wenn Du ein gutes System für eine PV Überschussladesteuerung suchst, schau Dir mal das Projekt EVCC  [https://evcc.io/](url) an. Es läuft bei mir super auf einem Raspberry PI. Super Steuerung des Ladestroms bei PV Überschuss, auch Auto Laden mit VX3 Batterieunterstützung.

Die Daten von der VX3 (PV Werte und Batterie) habe ich in EVCC über open3e mit MQTT eingebunden, ebenfalls Energiewerte E380 über E3onCAN mit MQTT. Deine Mennekes Wallbox sollte schon bei EVCC gelistet sein und kann von dort gesteuert werden.

**Comment by fanste** [comment]
Hey,

Ich würde dieses Thema gerne noch einmal aufgreifen. Nachdem bei uns gestern das Problem bestand, dass die ViCare nicht voll auf das System kam und deshalb die Wallbox nicht auf Schnellladen statt Grünladen gestellt werden konnte. Etwas blöde, wenn das Auto leer ist.

Daher wäre ich an einer Möglichkeit interessiert, dass lokal mit z. B. Home Assistant über open3e hinzubekommen.

Meine Anlage besteht aus VX3, 250-A (Master), Viessmann Wallbox (die Charge Control) und dem E380.

Im depictSystem via HA Addon von Dezember konnte ich aber nichts mit `wallbox`,  `vehicle`, `car` finden.

Kann ich hier irgendwie unterstützen, um herauszufinden, ob es entsprechende Datenpunkte für die Konfiguration der Wallbox gibt? Oder sind diese sogar schon bekannt? 

Die Energiedaten bekäme man ja über die Modbus Schnittstelle. Doch Grünstrom oder Schnellladen, das kann ich darüber vermutlich nicht beeinflussen.

**Reply by MyHomeMyData** [reply]
Die Wallbox ist per Ethernet eingebunden und hängt nicht am CAN-Bus. Ist das korrekt? Dann würde es mich sehr wundern, wenn das System Daten der Wallbox auf dem CAN-Bus kommunizieren würde. Also ist open3e hier nicht im Spiel.

Da kannst schauen, ob mit HA über Modbus Lösungen existieren. Kennst Du [evcc](https://evcc.io/)? Wahrscheinlich solltest Du aber nicht zwei Systeme (Viessmann One Base und ein zusätzliches System wie evcc) parallel betreiben.

**Reply by fanste** [reply]
Irgendwie bin ich wohl vorhin in der Antwort verrutscht, sorry. 
(Kontext: https://github.com/open3e/open3e/discussions/140#discussioncomment-15599404) 


[Diese](https://github.com/search?q=repo%3Aopen3e%2Fopen3e+vehicle&type=code) Suche zeigt einige Parameter zum Thema "Vehicle" und dort z. B. die DIDs, `3053 VehicleChargingStationControlSettings` oder `3054 VehicleChargingStationControlStatus`

**Comment by HDMatGIT** [comment]
Hallo, MyHomeMyData hat Recht. Ich habe mich schon lange bezüglich der Wallbox (Mennekes) von der Viessmann App erfolgreich verabschiedet. Soweit ich mich entsinnen kann, brauchte ich den Service für meine PV Anlage um die Wallbox zu trennen, bzw. aus der App zu entfernen. Da bin ich mir aber nicht mehr ganz sicher.
Bei mit läuft auf einem Raspberry der Adapter für den Can-Bus und holt die Daten von der Viessmann Anlage und parallel EVCC, einfach ein tolles Tool. Das könnte man natürlich auch in HA oder openHAB (wie bei mir) einbinden. Das hatte ich mal begonnen, sehe aber keine Vorteile drin. EVCC bekommt bei mir dann auch die flexiblen Stromtarife von Tibber, damit kann ich alles optimal steuern. Wie gesagt, die Integration davon in openHAB oder HA ist dann optische Spielerei, zumindest in meinen Augen.

**Reply by fanste** [reply]
Genau, die hängt nicht am CAN, aber die Einstellungen für Solar- oder Schnellladen kommen seitens Viessmann via TCU der Vitocal über die App. Also muss man das irgendwie parametrisieren können. TCU ist die Vermutung, weil damals dort ein Update eingespielt werden musste, um eine Funktion für die Wallbox verfügbar zu machen. 

Ich habe im Source von open3e Parameter gefunden, die darauf hindeuten, die bei mit im depict aber nicht vorkommen und auch nicht ansteuern kann. Sind die ggfs. nur auf dem internen Bus verfügbar? Gibt es einen Nachteil, wenn ich vom externen auf den internen wechsle? Oder sollte ich dann lieber mit 2 Adaptern arbeiten?

https://github.com/search?q=repo%3Aopen3e%2Fopen3e+vehicle&type=code

Z. B. 3053 oder 3054.


EVCC will ich aktuell nicht noch zusätzlich einrichten, da ansonsten das Zusammenspiel bisher funktioniert. (abgesehen von PV Überschuss in Wärme, aufgrund von Bugs in der 250-A)

**Reply by MyHomeMyData** [reply]
> Ich habe im Source von open3e Parameter gefunden, die darauf hindeuten, die bei mit im depict aber nicht vorkommen und auch nicht ansteuern kann. Sind die ggfs. nur auf dem internen Bus verfügbar? 

Welche Parameter meinst Du?

**Reply by MyHomeMyData** [reply]
@philippoo66 Phil, wie kommt die Datei `DidEnums.txt` zustande und was sagen die Infos aus? In `Open3Edatapoints.py` finde ich keine Datenpunkte, die "Vehicle" enthalten.

@fanste Mir sind bisher keine Viessmann-Datenpunkte bekannt, die auf dem internen Bus gelesen werden können, aber nicht auf dem externen. Manche Infos von anderen Devices, z.B. E380, sind auf dem externen Bus verfügbar, aber nicht auf dem internen. Man kann also nicht ausschließen, dass es auch den umgekehrten Fall gibt. Ich sag mal, Versuch macht kluch.

Unterschiede gibt es auf jeden Fall beim Schreiben von Datenpunkten. Da sind uns Dids bekannt, die über den internen Bus verändert werden können (Stichwort Service 77), aber nich über den externen.

Wenn Du den E380 nicht auswertest, sind mir keine Nachteile des internen Busses bekannt. Falls Du zwei USB2CAN-Adapter parallel betreibst, musst Du auf die feste Zuordnung im Linux-System achten. Sonst gerät das evtl. beim nächsten Reboot durcheinander.

**Reply by philippoo66** [reply]
> wie kommt die Datei DidEnums.txt zustande und was sagen die Infos aus? 

die kommt aus den Viguide sourcen. die Namen werden in den datapoinnts_6yz benutzt, wenn ein entsprechender DID gefunden wird und es noch keinen Eintrag in datapoints.py gibt.

**Reply by MyHomeMyData** [reply]
@philippoo66 Das sollten wir transparent machen, denn innerhalb des Repos ist `DidEnums.txt` die einzige Stelle mit diesen Infos. Können wir die Infos aus ViGuide in `Open3Edatapoints.py` integrieren? Am besten per Tool?

**Reply by philippoo66** [reply]
@MyHomeMyData  das Problem ist ja, dass solange niemand so einen DID ausgelesen hat (und uns das mitgeteilt) kennen wir die Länge nicht. Um es in Open3Edatapoints.dataIdentifiers zu integrieren brauchen wir aber zumindest eine Länge...

**Reply by philippoo66** [reply]
wir könnten natürlich relativ problemlos ein zweites dict in Open3Edatapoints.py beipacken, aber ist das wirklich sinnvoll? Wenn dann mal wieder jemand durch eine neue APK geht und die Liste aktualisiert, sieht es so aus, als hätte sich Open3Edatapoints verändert, aber die _eigentlichen_ dataIdentifiers sind gleich geblieben... Ich finde wir sollten das nicht unbedingt durcheinandermixen!?

Es hat ja auch keinen weiteren Nutzen. ausser depict kann niemand/nix was mit der Information anfangen.

**Reply by MyHomeMyData** [reply]
@philippoo66 Ja, dann macht es keinen Sinn. Dass die Länge unbekannt ist, war mir nicht klar.

**Reply by fanste** [reply]
@philippoo66 Wie kann ich hier unterstützen? Einfach an meiner Anlage einmal den intern Bus der 250-A anschließen und ein depcit laufen lassen? Welche Informationen sind davon für euch dann sinnvoll? Oder muss/sollte ich anders vorgehen, um benötigte Informationen bereitzustellen?

**Reply by philippoo66** [reply]
unterstützen kannst du, indem du depictSystem **mit der Option -s** durchlaufen lässt und uns dann alle datapoints...py, die devices.json und alle virtdata_....txt schickst, und eine kleine Note um was für eine Anlage es sich handelt und an welchem Bus (also ggf. 2 Datensätze).

Bekommst du denn die Vehicle DIDs?

**Reply by fanste** [reply]
Ich bin aktuell mit dem externen Bus (zwischen 250-A und VX3) verbunden. Da bekomme ich sie nicht.

An den internen Bus habe ich mich noch nicht herangetraut, um zu schauen, ob diese nur dort verfügbar sind.
Versuche ich, sobald ich mehr wie nur 2h Luft habe.

**Comment by HDMatGIT** [comment]
Sorry, da kann ich leider nichts beitragen. Ich müsste mich überhaupt erst mal wieder mit open3e beschäftigen, dazu fehlt mir leider die Zeit. Wie bereits gesagt hat mich die mangelnde Funktionalität der Viessmann App, insbesondere mit der Wallbox, schnell dazu gebracht, EVCC zu installieren und die Daten der PV Anlage in eine schöne Animation in meinem openHAB umzusetzen. Ich nutze die App eigentlich kaum noch. Um die Zählerdaten zu bekommen nutze ich übrigens noch E3onCAN, denn als ich das alles umgesetzt habe, hat der can-Bus des Inverters keine brauchbaren Daten geliefert, also nicht nur für die Wallbox. Das läuft bei mir nun etwa 2 Jahre sehr erfolgreich und gut. Wenn sich kein Zwang auftut das alles wieder anzufassen, will ich das auch nicht tun, ganz gemäß dem Motto, "never change a running system"!
